render/wgpu/shaders/bitmap.wgsl

```diff
@@ -42,14 +42,13 @@ fn main_fragment(in: VertexOutput) -> @location(0) vec4<f32> {
     if( color.a > 0.0 ) {
         color = vec4<f32>(color.rgb / color.a, color.a);
         color = color * colorTransforms.mult_color + colorTransforms.add_color;
-        // Apply 'saturate' *after* we re-multiply. If we get a non-premultiplied
-        // color like (1.0, 1.0. 1.0, 0.5) from a Stage3D texture, we want
-        // this shader to have no effect if we're applying a no-op colorTransforms.
-        // By applying the saturation after dividing and multiply by the alpha,
-        // we avoid saturing if the intermediate 'unmultiplied' color is temporarily
-        // out of range.
-        let alpha = saturate(color.a);
-        color = vec4<f32>(saturate(color.rgb * alpha), alpha);
+        #if early_saturate == true
+            color = saturate(color);
+        #endif
+        color = vec4<f32>(color.rgb * color.a, color.a);
+        #if early_saturate == false
+            color = saturate(color);
```

Comment:
Won't the fragment shader output automatically be clamped? Or can this write to a floating-point buffer?

Reply:
Assuming the texture format is something non-float like RGBA8, the output should get automatically clamped, but I do see there's a `Context3DTextureFormat.RGBA_HALF_FLOAT`, so maybe the saturate would be needed for that case regardless.

Reply:
It'll depend if we're using a *NORM texture or not. For now, we only use NORM and so everything is saturated after writing, and we don't need to saturate here. It's not really going to impact anything either way though, so I'm fine with it!